src/string_constraints.cpp

```
@@ -0,0 +1,72 @@
+        return bottom();
+    }
+    StringInvariant res = top();
+    for (const std::string& cst : this->value()) {
+        if (b.is_bottom() || !b.contains(cst)) {
+            res.maybe_inv->insert(cst);
+        }
+    }
+    return res;
+}
+
+StringInvariant StringInvariant::operator+(const StringInvariant& b) const {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)